code/modules/power/supermatter/supermatter.dm

```diff
@@ -755,6 +755,13 @@ GLOBAL_DATUM(main_supermatter_engine, /obj/machinery/power/supermatter_crystal)
 		var/mob/living/user = AM
 		if(user.status_flags & GODMODE)
 			return
+		if(ishuman(AM))
+			var/mob/living/carbon/human/proofed = user
+			if(proofed.dna && SUPERMATTERPROOF in proofed.dna.mutations)
```

Comment:
The power itself is way too niche in my opinion, but if you're keeping it consider making this check based on a trait instead, so other atoms can be set as supermatter immune as needed.